### ISSUE TITLE
Docker compose fails to pull podman images

### ISSUE THREAD

**Issue by SureshGota** [containers/podman#18843]
### Issue Description

I am working on a windows and have installed Podman. Inside the default WSL created by Podman I have installed docker compose.

Here's my compose file,
```
version: "3.9"
services:
    configurationsService:
        image: "localhost/configurations"
        hostname: configurationsService
        ports:
          - "51005:51005"
        restart: always
```

```
**Docker Compose** version v2.15.1

**Podman Version**
Client:       Podman Engine
Version:      4.5.0
API Version:  4.5.0
Go Version:   go1.19.7
Built:        Fri Apr 14 21:12:56 2023
OS/Arch:      linux/amd64
```

### Steps to reproduce the issue

I have built the images using `podman build` command and I can see them when I list them as `podman images`. I am also able to run the images using `podman run` command. 
However when I run `docker-compose up` it fails while trying to pull the image
with the error `initializing source docker://localhost/configurations:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp 127.0.0.1:443: connect: connection refused`

if I remove the localhost prefix from my image name it tries to connect to docker.io and fails there as well.

### Describe the results you received

Describe the results you received

### Describe the results you expected

Fails to pull the images from local machine.

### podman info output

```yaml
NA
```

### Podman in a container

No

### Privileged Or Rootless

Privileged

### Upstream Latest Release

Yes

### Additional environment details

Additional environment details

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by Luap99** [comment]
@vrothberg PTAL

**Comment by vrothberg** [comment]
Thanks for reaching out, @SureshGota!

Do I understand you correctly that the `localhost/configurations` is present in the local image store?  With exactly this name/tag?

**Comment by SureshGota** [comment]
@vrothberg vrothberg yes its the local image store. 

The same yaml file works fine with dockerD and docker image store, But with podman & docker compose it fails.

when I prefix with localhost/imageName it fails with the error _`initializing source docker://localhost/configurations:latest: pinging container registry localhost: Get "https://localhost/v2/": dial tcp 127.0.0.1:443: connect: connection refused`_

If I remove the prefix localhost and try, this time it fails while connecting to docker.io but its an authentication related failure.

**Comment by vrothberg** [comment]
Thanks, @SureshGota !  I will investigate and report back soon.

**Comment by vrothberg** [comment]
I tried with Podman v4.5.1 and am unable to reproduce.  Ultimately, the error from Podman indicates that the image is not present in the local storage (or cannot be found).  Can you share the output of `podman images`?  Are you running `podman` and `docker-compose` under the same user?

**Comment by SureshGota** [comment]
@vrothberg Another critical info that I would like to add, I ran a sample docker-compose to run ngnix and redis image that was stored on docker.io. This ran successfully, however when I did `podman images` I couldn't find these images but `docker-compose images`  showed me both ngnix and redis images which were removed only after I did `docker-compose down`.

**Comment by SureshGota** [comment]
@vrothberg Did you install docker-compose manually? I did this step additionally.

I am running both as same user but for running docker-compose I need to sudo.

![image](https://github.com/containers/podman/assets/6790860/11ac4528-d0e1-499b-b712-9a525edca1aa)

**Comment by vrothberg** [comment]
> [...] but for running docker-compose I need to sudo

In this case, `docker-compose` will probably talk to the rootfull podman not the one of your non-root user.  If you a `sudo podman images`, the `localhost/....` one will not show up.

Why do you need to run `docker-compose` as root?  Dropping the `sudo` or making the image available for root should resolve your issue.

**Comment by SureshGota** [comment]
Good catch. Yes, when you do `sudo podman images `I do not see `localhost/...`

The reason I run docker-compose with sudo is because of this error `permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/json?all=1&filters=%7B%22label%22%3A%7B%22com.docker.compose.project%3Dcode_repo%22%3Atrue%7D%7D": dial unix /var/run/docker.sock: connect: permission denie`

**Comment by vrothberg** [comment]
If you want to point `docker-compose` to the rootless Podman socket, you can point it via an env variable: `export DOCKER_HOST=unix://$XDG_RUNTIME_DIR/podman/podman.sock`

Is it working for you?

**Comment by vrothberg** [comment]
> I have added the below config in my compose file but looks like the containers are not able to access the host service. Any advice on this?

I am not sure if that works today.  @Luap99, do you?

We may need to create a separate issue for it.

**Comment by Luap99** [comment]
There is https://github.com/containers/podman/issues/14390 for that.
Right now we only have our own `host.containers.internal` entry which is added by default.

I am closing this one as the pull thing is resolved.
